pkg/sypgp/sypgp.go

```diff
@@ -376,144 +479,136 @@ func GetPassphrase(message string, retries int) (string, error) {
 	conf := &packet.Config{RSABits: 4096, DefaultHash: crypto.SHA384}
 
-	if err = PathsCheck(); err != nil {
-		return
+	entity, err := openpgp.NewEntity(name, comment, email, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// encrypt private key
+	if err = EncryptKey(entity, passphrase); err != nil {
+		return nil, err
+	}
+
+	// Store key parts in local key caches
+	if err = StorePrivKey(entity); err != nil {
+		return nil, err
+	}
+
+	if err = StorePubKey(entity); err != nil {
+		return nil, err
+	}
+
+	return entity, nil
+}
+
+// GenKeyPair generates an OpenPGP key pair and store them in the sypgp home folder
+func GenKeyPair(keyServiceURI string, authToken string) (*openpgp.Entity, error) {
+	if err := PathsCheck(); err != nil {
+		return nil, err
 	}
 
 	name, err := AskQuestion("Enter your name (e.g., John Doe) : ")
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	email, err := AskQuestion("Enter your email address (e.g., john.doe@example.com) : ")
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	comment, err := AskQuestion("Enter optional comment (e.g., development keys) : ")
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	// get a password
 	passphrase, err := GetPassphrase("Enter a passphrase : ", 3)
 	if err != nil {
-		return
+		return nil, err
 	}
 
-	fmt.Printf("Generating Entity and OpenPGP Key Pair...")
-	entity, err = openpgp.NewEntity(name, comment, email, conf)
-	if err != nil {
-		return
-	}
-	fmt.Printf("done\n")
+	fmt.Printf("Generating Entity and OpenPGP Key Pair... ")
 
-	// encrypt private key
-	if err = EncryptKey(entity, passphrase); err != nil {
-		return
+	entity, err := genKeyPair(name, comment, email, passphrase)
+	if err != nil {
+		return nil, err
 	}
 
-	// Store key parts in local key caches
-	if err = StorePrivKey(entity); err != nil {
-		return
-	}
-	if err = StorePubKey(entity); err != nil {
-		return
-	}
+	fmt.Printf("done\n")
 
 	// Ask to push the new key to the keystore
-	pushKeyQ, err := AskQuestion("Would you like to push it to the keystore? [Y,n] : ")
-	if err != nil {
-		return
-	}
+	ans, err := askYNQuestion("y", "Would you like to push it to the keystore? [Y,n] ")
+	switch {
+	case err != nil:
+		fmt.Printf("Not pushing newly created key to keystore: %s\n", err)
 
-	if pushKeyQ == "" || pushKeyQ == "y" || pushKeyQ == "Y" {
+	case ans == "y":
 		err = PushPubkey(entity, keyServiceURI, authToken)
 		if err != nil {
-			return
+			fmt.Printf("Failed to push newly created key to keystore: %s\n", err)
+		} else {
+			fmt.Printf("Key successfully pushed to: %s\n", keyServiceURI)
 		}
-		fmt.Printf("Key successfully pushed to: %v\n", keyServiceURI)
+
+	default:
+		fmt.Printf("NOT pushing newly created key to: %s\n", keyServiceURI)
 	}
-	fmt.Printf("Done.\n")
 
-	return
+	return entity, nil
 }
 
 // DecryptKey decrypts a private key provided a pass phrase
 func DecryptKey(k *openpgp.Entity, message string) error {
+	if !k.PrivateKey.Encrypted {
+		return errNotEncrypted
```

Comment:
Why return a error if not encrypted?

Reply:
my expectation is that if you are calling the DecryptKey function it's because you think the key is encrypted. If the key is not encrypted, why are you calling DecryptKey?

Reply:
What if you remove the `if !k.PrivateKey.Encrypted {...}`, then when we want to decrypt a key later we can do:

```go
if k.PrivateKey.Encrypted {
    err := sypgp.DecryptKey(entity, pass)
    if err != nil {
        sylog.Fatalf("Unable to decrypt key: %s", err)
    }
}
```

Of course you can check `k.PrivateKey.Encrypted` anyway, but then `k.PrivateKey.Encrypted` will be called twice.